src/main/java/kitchenpos/menu/application/MenuValidator.java

```diff
@@ -0,0 +1,95 @@
+package kitchenpos.menu.application;
+
+import java.math.BigDecimal;
+import java.util.List;
+import java.util.Map;
+import java.util.NoSuchElementException;
+import java.util.stream.Collectors;
+import kitchenpos.common.domain.Price;
+import kitchenpos.common.domain.Quantity;
+import kitchenpos.menu.dto.MenuProductRequest;
+import kitchenpos.menu.dto.MenuRequest;
+import kitchenpos.menugroup.domain.MenuGroupRepository;
+import kitchenpos.product.domain.Product;
+import kitchenpos.product.domain.ProductRepository;
+import org.springframework.stereotype.Component;
+
+@Component
+public class MenuValidator {
+    private final ProductRepository productRepository;
+    private final MenuGroupRepository menuGroupRepository;
+
+    public MenuValidator(ProductRepository productRepository, MenuGroupRepository menuGroupRepository) {
+        this.productRepository = productRepository;
+        this.menuGroupRepository = menuGroupRepository;
+    }
+
+    public void validate(MenuRequest request) {
+        validateExistingMenuGroup(request);
+        validateExistingProduct(request);
+        validateAmount(request);
+
+    }
+
+    private void validateExistingProduct(MenuRequest request) {
+        List<MenuProductRequest> menuProducts = request.getMenuProductRequests();
+        List<Product> findProducts = findAllProductsInMenuProduct(menuProducts);
+
+        if (menuProducts.size() != findProducts.size()) {
+            throw new IllegalArgumentException("시스템에 등록되어 있지 않은 메뉴 상품이 있습니다.");
+        }
+    }
+
+    private void validateExistingMenuGroup(MenuRequest request) {
+        if (!menuGroupRepository.existsById(request.getMenuGroupId())) {
+            throw new NoSuchElementException("메뉴그룹이 시스템에 존재 하지 않습니다");
+        }
+    }
+
+    private void validateAmount(MenuRequest request) {
+        List<MenuProductRequest> menuProducts = request.getMenuProductRequests();
+
+        List<Product> findProducts = findAllProductsInMenuProduct(menuProducts);
+        Map<Long, Price> productIdPrices = makeProductIdPriceMap(findProducts);
+
+        Price total = calculateTotalAmount(menuProducts, productIdPrices);
+
+        if (new Price(BigDecimal
+                .valueOf(request.getPrice()))
```

Comment:
```suggestion
        if (new Price(request.getPrice())
```

Price를 생성할 수 있는 방법을 확장해보면 어떨까요?

Reply:
생성자 추가하겠습니다